packages/@react-aria/color/src/useColorArea.ts

```diff
@@ -0,0 +1,578 @@
+        };
+        background.gradientStyles = {
+          /* the foreground represents the red channel as a linear gradient from min to max,
+          with the blue channel maximized, adjusted by the green channel value. */
+          backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(0,${zValue},255),rgb(255,${zValue},255))`,
+          /* the foreground gradient is masked by a perpendicular linear gradient from black to white */
+          'WebkitMaskImage': maskImage,
+          maskImage
+        };
+        break;
+      }
+      case 'blue': {
+        dir = xChannel === 'red';
+        maskImage = `linear-gradient(to ${orientation[Number(!dir)]}, transparent, #000)`;
+        background.colorAreaStyles = {
+          /* the background represents the red channel as a linear gradient from min to max,
+          with the green channel minimized, adjusted by the blue channel value. */
+          backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(0,0,${zValue}),rgb(255,0,${zValue}))`
+        };
+        background.gradientStyles = {
+          /* the foreground represents the red channel as a linear gradient from min to max,
+          with the green channel maximized, adjusted by the blue channel value. */
+          backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(0,255,${zValue}),rgb(255,255,${zValue}))`,
+          /* the foreground gradient is masked by a perpendicular linear gradient from black to white */
+          'WebkitMaskImage': maskImage,
+          maskImage
+        };
+        break;
+      }
+    }
+    return background;
+  };
+
+  let background = generateBackground();
+  let ariaRoleDescription = messages.getStringForLocale('twoDimensionalSlider', locale);
+  
+  let {visuallyHiddenProps} = useVisuallyHidden();
+  visuallyHiddenProps.style = mergeProps(
+    visuallyHiddenProps.style,
+    {
+      opacity: '0.0001',
+      width: '100%',
+      height: '100%',
+      pointerEvents: 'none'
+    }
+  );
+
+  return {
+    colorAreaProps: {
+      ...colorAriaLabellingProps,
+      ...colorAreaInteractions,
+      role: 'group',
+      style: {
+        position: 'relative',
+        touchAction: 'none',
+        ...background.colorAreaStyles
+      }
+    },
+    gradientProps: {
+      style: {
+        touchAction: 'none',
+        ...background.gradientStyles
+      }
+    },
+    thumbProps: {
+      ...thumbInteractions,
+      style: {
+        position: 'absolute',
+        left: `${x * 100}%`,
+        top: `${y * 100}%`,
+        transform: 'translate(0%, 0%)',
+        touchAction: 'none'
+      }
+    },
+    xInputProps: {
+      ...inputLabellingProps,
+      ...visuallyHiddenProps,
+      type: 'range',
+      min: state.value.getChannelRange(xChannel).minValue,
+      max: state.value.getChannelRange(xChannel).maxValue,
+      step: String(step),
```

Comment:
Just a thought, and lemme know if this was already discussed prior, but would it make sense to have two different `step` props here since the x and y directions can/will differ in the range of values they support?

Reply:
Probably makes sense to have a xChannelStep and a yChannelStep, or to set step using value returned from `getChannelRange`.